saml2/saml2-service-provider/core/src/main/java/org/springframework/security/saml2/provider/service/web/authentication/logout/OpenSamlLogoutRequestHandler.java

```diff
@@ -0,0 +1,198 @@
+import org.opensaml.saml.saml2.core.NameID;
+import org.opensaml.saml.saml2.core.impl.LogoutRequestUnmarshaller;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.saml2.Saml2Exception;
+import org.springframework.security.saml2.core.OpenSamlInitializationService;
+import org.springframework.security.saml2.core.Saml2Error;
+import org.springframework.security.saml2.core.Saml2ErrorCodes;
+import org.springframework.security.saml2.core.Saml2ResponseValidatorResult;
+import org.springframework.security.saml2.provider.service.authentication.Saml2Authentication;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistration;
+import org.springframework.security.saml2.provider.service.web.RelyingPartyRegistrationResolver;
+import org.springframework.security.saml2.provider.service.web.authentication.logout.OpenSamlVerificationUtils.VerifierPartial;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link LogoutHandler} that handles SAML 2.0 Logout Requests received from a SAML 2.0
+ * Asserting Party.
+ *
+ * @author Josh Cummings
+ * @since 5.5
+ */
+public final class OpenSamlLogoutRequestHandler implements LogoutHandler {
+
+	static {
+		OpenSamlInitializationService.initialize();
+	}
+
+	private final RelyingPartyRegistrationResolver relyingPartyRegistrationResolver;
+
+	private final ParserPool parserPool;
+
+	private final LogoutRequestUnmarshaller unmarshaller;
+
+	/**
+	 * Constructs a {@link OpenSamlLogoutRequestHandler} from the provided parameters
+	 * @param relyingPartyRegistrationResolver the
+	 * {@link RelyingPartyRegistrationResolver} from which to derive the
+	 * {@link RelyingPartyRegistration}
+	 */
+	public OpenSamlLogoutRequestHandler(RelyingPartyRegistrationResolver relyingPartyRegistrationResolver) {
+		this.relyingPartyRegistrationResolver = relyingPartyRegistrationResolver;
+		XMLObjectProviderRegistry registry = ConfigurationService.get(XMLObjectProviderRegistry.class);
+		this.parserPool = registry.getParserPool();
+		this.unmarshaller = (LogoutRequestUnmarshaller) XMLObjectProviderRegistrySupport.getUnmarshallerFactory()
+				.getUnmarshaller(LogoutRequest.DEFAULT_ELEMENT_NAME);
+	}
+
+	/**
+	 * Processes the SAML 2.0 Logout Request received from the SAML 2.0 Asserting Party.
+	 *
+	 * By default, verifies the signature, validates the issuer, destination, and user
+	 * identifier.
+	 *
+	 * If any processing step fails, a {@link Saml2Exception} is thrown, stopping the
+	 * logout process
+	 * @param request the HTTP request
+	 * @param response the HTTP response
+	 * @param authentication the current principal details
+	 */
+	@Override
+	public void logout(HttpServletRequest request, HttpServletResponse response, Authentication authentication) {
```

Comment:
I believe this would create `OpenSamlLogoutRequestFilter`. In the past, I think I understood that pushing library support up to the filter level was undesirable. Not sure if I remembered that correctly.

If it's reasonable to introduce a class like that, I believe it's possible to embed the behavior into a filter and break it out later on as folks can replace the filter instead of the component. Personally as a user, I'd prefer changing out a component though so that I don't need to consider the proper filter order.

Reply:
I'm also a little concerned about placing OpenSAML code in filters since that could potentially introduce duplicate filters down the road, say if OpenSAML introduces breaking changes.

Reply:
Addressed by introducing `Saml2LogoutRequestAuthenticator` and `Saml2LogoutResponseAuthenticator` and accompanying implementations.

Reply:
I landed on "authenticator" because we tend to use verifier for checking the signature, validator for checking other aspects, and this component does both verifying and validating. JwtDecoder is another example of a component that both verifies the signature and validates other things. An authentication provider exercises the decoder.

The reason this component isn't a decoder is that no decoded version of the logout request is returned to the caller.

Though I don't see the name "authenticator" used commonly in Spring Security, I feel it's the best option since both verifier and validator since this component does both those things.